MeshGeoToolsLib/BoundaryElementsAlongPolyline.cpp

```diff
@@ -57,22 +57,50 @@ BoundaryElementsAlongPolyline::BoundaryElementsAlongPolyline(
                 if (edge != new_edge)
                     delete edge;
                 _boundary_elements.push_back(new_edge);
+                _bulk_ids.emplace_back(std::make_pair(e->getID(), i));
             } else {
                 delete edge;
             }
         }
     }
 
-    // sort picked edges according to a distance of their first node along the polyline
-    std::sort(_boundary_elements.begin(), _boundary_elements.end(),
-            [&](MeshLib::Element*e1, MeshLib::Element*e2)
-            {
-                std::size_t dist1 = std::distance(node_ids_on_poly.begin(),
-                    std::find(node_ids_on_poly.begin(), node_ids_on_poly.end(), e1->getNodeIndex(0)));
-                std::size_t dist2 = std::distance(node_ids_on_poly.begin(),
-                    std::find(node_ids_on_poly.begin(), node_ids_on_poly.end(), e2->getNodeIndex(0)));
-                return (dist1 < dist2);
-            });
+    // zip _boundary_elements and _bulk_element_ids
+    std::vector<std::pair<MeshLib::Element*, std::pair<std::size_t, unsigned>>>
+        zipped_data;
+    zipped_data.reserve(_boundary_elements.size());
+    std::transform(_boundary_elements.cbegin(), _boundary_elements.cend(),
+                   _bulk_ids.cbegin(), std::back_inserter(zipped_data),
+                   [](MeshLib::Element* boundary_element,
+                      std::pair<std::size_t, unsigned> const& bulk_ids) {
+                       return std::make_pair(boundary_element, bulk_ids);
+                   });
+
+    // sort picked edges according to a distance of their first node along the
+    // polyline
```

Comment:
Is that necessary to speed up some subsequent linear searches?

Reply:
The boundary elements have been sorted in the previous version of the implementation, too. I think the sort was necessary in OGS-5 for some reason. I'm not sure if it is needed anymore in OGS-6.

Reply:
Probably currently it's too much work to check if that code is still needed. But could you please add a comment that the sorting might be unnecessary. Because otherewise somebody who will refactor that piece of code later one might be pretty puzzled.